src/test/java/org/jabref/logic/journals/AbbreviationTest.java

```diff
@@ -101,4 +105,13 @@ void testDefaultAndShortestUniqueAbbreviationsAreSame() {
         Abbreviation abbreviation = new Abbreviation("Long Name", "L N");
         assertEquals(abbreviation.getAbbreviation(), abbreviation.getShortestUniqueAbbreviation());
     }
+
+    @Test
```

Comment:
Nitpick: Parametrized tests are always welcome 😇

Reply:
Done in 1c4dce3
I have to admit I don't exactly see the benefit of having it as a parameterized test (did I do it wrong?), but I took the opportunity to get acquainted with the concept and JUnit's implementation and implemented it anyway.

Reply:
Agreed, (and thank you for updating them anyway!) it doesn't add all that much here (which is partly why I considered it a nitpick, I'd have accepted the old code as well). My thoughts were

1. We generally prefer [one thing tested in each test](https://devdocs.jabref.org/getting-into-the-code/code-howtos#general-hints-on-tests). That might be interpreted as splitting the test into "long", "abbreviation" and "shortest unique abbreviation" tests. Making the test parameterized makes this point moot
2. All the other tests in the file seem like they can be transformed into parameterized tests, if someone decides to update the old test, it'll be easier for them if your tests are already parameterized. As it appears like we are moving more towards parameterized tests I believe it is a reasonable chance it will happen. I'll probably try doing those kind of rewrites (where I think it makes some sense) with my future PRs (in case I end up being the person doing the rewriting, thank you in advance 😛 )
3. It will be easier for someone else to add/remove/modify test cases (thank you for adding the empty tests as well)

---

Looking through the tests again, shouldn't these be in a new test file? `AbbreviationViewModelTest.java`? You are testing a method of the ViewModel, not `Abbreviation.java`.

---

I'd probably consider splitting the test method into two, perhaps,

* `containsCaseIndependentDoesNotContain`
* `containsCaseIndependentContains`

But that might be in deep nitpicking territory and probably more a matter of taste. A note about the naming, we are trying to avoid prefixing tests with 'test...', despite all other tests in that file having that naming convention.

Reply:
Thanks for the clarification!
Moving AbbreviationViewModelTest.java makes sense. Since we are testing on the view model, I guess it would make even more sense to move that to the gui package, maybe in org.jabref.gui.journals?

Reply:
I am not completely sure that I understand you,

1. create `src/test/java/org/jabref/gui/preferences/journals/AbbreviationViewModelTest.java`
2. copy the tests you created to the new file

have I understood you correctly?

(on a side note, if you are using IntelliJ and viewing `AbbreviationViewModel` you can use the [code | generate](https://www.jetbrains.com/help/idea/generating-code.html) -> `Test...` and the default settings should generate it in the right place)

Reply:
I was just hesistant since there was no gui/preferences directory. But obviously, that would be the best path.
I used IntelliJ's code generator now, thanks for the hint!